src/components/common/Extensions/UserHub/partials/NotificationsTab/partials/Notification.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  type IRemoteNotification,
+  useNotification,
+} from '@magicbell/react-headless';
+import clsx from 'clsx';
+import React, { type FC } from 'react';
+import { defineMessages } from 'react-intl';
+
+import { ADDRESS_ZERO } from '~constants';
+import { useGetColonyActionQuery } from '~gql';
+import { type Notification as NotificationInterface } from '~types/notifications.ts';
+import { formatText } from '~utils/intl.ts';
+import ColonyAvatar from '~v5/shared/ColonyAvatar/index.ts';
+import RelativeDate from '~v5/shared/RelativeDate/index.ts';
+
+import NotificationMessage from './NotificationMessage.tsx';
+
+const displayName = 'common.Extensions.UserHub.partials.Notification';
+
+const MSG = defineMessages({
+  unknownAction: {
+    id: `${displayName}.unknownAction`,
+    defaultMessage: 'Unknown action',
+  },
+  unknownColony: {
+    id: `${displayName}.unknownColony`,
+    defaultMessage: 'Unknown colony',
+  },
+});
+
+interface NotificationProps {
+  notification: NotificationInterface;
+}
+
+const Notification: FC<NotificationProps> = ({ notification }) => {
+  const { markAsRead } = useNotification(notification as IRemoteNotification);
+  const { data: actionData, loading: loadingAction } = useGetColonyActionQuery({
+    variables: {
+      transactionHash: notification.customAttributes?.transactionHash || '',
+    },
+    skip: !notification.customAttributes?.transactionHash,
+  });
+
+  const action = actionData?.getColonyAction;
+  const colony = action?.colony;
+
+  const handleNotificationClicked = () => {
+    if (!notification.readAt) {
+      markAsRead();
+    }
+  };
+
+  return (
+    <li className="w-full py-3.5 first-of-type:pt-0">
+      <button
+        className={clsx('relative  flex w-full gap-2 text-left', {
+          skeleton: loadingAction,
+        })}
+        onClick={handleNotificationClicked}
+        type="button"
+      >
+        <ColonyAvatar
+          size={20}
+          colonyAddress={colony?.colonyAddress || ADDRESS_ZERO}
```

Comment:
Just a heads up that Address Zero renders the blockie as a full black image. So that's a handy way of knowing you've hit the fallback :)